Content/Items/Misc/Accessories.TaintedGreataxe.cs

```diff
@@ -0,0 +1,486 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Content.Items.Potions;
+using StarlightRiver.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.Graphics.Effects;
+using StarlightRiver.Helpers;
+using Terraria.Audio;
+using System.IO;
+using Terraria.GameContent;
+using StarlightRiver.Core.Systems;
+
+namespace StarlightRiver.Content.Items.Misc
+{
+	public class TaintedGreataxe : CursedAccessory
+	{
+		public override string Texture => AssetDirectory.MiscItem + Name;
+
+		public TaintedGreataxe() : base(ModContent.Request<Texture2D>(AssetDirectory.MiscItem + "TaintedGreataxe").Value) { }
+
+		public override void SetStaticDefaults()
+		{
+			Tooltip.SetDefault("Cursed\nSummons an Ethereal Greataxe, which embeds itself in enemies who were just critically striked\nThe enemy that the Greataxe embeds itself in becomes Tainted\n<right> on the Greataxe whilst it is Embedded and it will un-embed itself");
+		}
+
+        public override void SafeUpdateEquip(Player Player)
+        {
+			if (Player.ownedProjectileCounts[ModContent.ProjectileType<TaintedGreataxeProjectile>()] < 1 && Player.whoAmI == Main.myPlayer)
+				Projectile.NewProjectileDirect(Player.GetSource_Accessory(Item), Player.Center, Vector2.One, ModContent.ProjectileType<TaintedGreataxeProjectile>(),
+					(int)Player.GetTotalDamage(DamageClass.Generic).ApplyTo(30), 2f, Player.whoAmI);
+
+			Player.GetModPlayer<TaintedGreataxePlayer>().HasGreatAxe = true;
+        }
+    }
+
+	class TaintedGreataxeProjectile : ModProjectile
+    {
+        public bool init = false;
+        private List<Vector2> oldPosition = new List<Vector2>();
+        private List<float> oldRotation = new List<float>();
+
+        public NPC targetNPC;
+        public bool Embedding;
+        public int EmbeddingCooldown;
+
+        public bool stickyAI;
+        public Vector2 offset = Vector2.Zero;
+        public int enemyWhoAmI; 
+
+        public Player Player => Main.player[Projectile.owner];
```

Comment:
lets change the name of this to ``Owner`` to match patterns used elsewhere in the mod